Lib/test/test_netrc.py

```diff
@@ -123,6 +123,19 @@ def test_security(self):
             os.chmod(fn, 0o622)
             self.assertRaises(netrc.NetrcParseError, netrc.netrc)
 
+    def test_file_not_found_in_home(self):
+        d = support.TESTFN
+        os.mkdir(d)
+        self.addCleanup(support.rmtree, d)
+        with support.EnvironmentVarGuard() as environ:
+            environ.set('HOME', d)
```

Comment:
> This doesn't test a new behavior

I think it does. It checks the _type_ of the exception raised, which is the whole point of this patch,
getting rid of the weird 'OSError' about HOME not set ...

> Needed a test for the case when HOME is not set.

Yeah I was not sure about this one. I've added a test where I monkeypatch `os.path.expanduser`, but I'm not sure it's useful. My guts tell me such a test is too close to the implementation.